Cython/Compiler/Naming.py

```diff
@@ -177,3 +178,18 @@
 
 def py_version_hex(major, minor=0, micro=0, release_level=0, release_serial=0):
     return (major << 24) | (minor << 16) | (micro << 8) | (release_level << 4) | (release_serial)
+
+class _py_code:
+    # This is in Naming.py to be able to abuse the UtilityCode "subsitute naming" mechanism.
```

Comment:
Can you imagine that some people on this planet would sense the word "hack" floating by when facing this code?

Reply:
In fairness this was the one bit of code I still understood 6 months later, which suggests it was probably the best bit.

I've replaced it with Tempita though.